airbyte-webapp/src/views/Connector/ConnectorCard/hooks/useOnHandleSubmit.tsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useState } from "react";
+
+import { Connector, ConnectorDefinition } from "core/domain/connector";
+import { ServiceFormValues } from "views/Connector/ServiceForm";
+
+import { CheckConnectionRead } from "../../../../core/request/AirbyteClient";
+import { useAnalyticsTrackFunctions } from "../useAnalyticsTrackFunctions";
+
+interface UseOnHandleSubmitHookProps {
+  setErrorStatusRequest: (error: Error | null) => void;
+  availableServices: ConnectorDefinition[];
+  formType: "source" | "destination";
+  testConnector: (v?: ServiceFormValues) => Promise<CheckConnectionRead>;
+  onSubmit: (values: ServiceFormValues) => Promise<void> | void;
+}
+
+interface UseOnHandleSubmitHookResult {
+  isFormSubmitting: boolean;
+  onHandleSubmit: (values: ServiceFormValues) => Promise<void>;
+  saved: boolean;
+}
+
+export const useOnHandeSubmit = (props: UseOnHandleSubmitHookProps): UseOnHandleSubmitHookResult => {
```

Comment:
I don't know if encapsulating this logic in a hook is actually helpful, and it feels like a slightly

I feel like it might be more readable to just put this onSubmit logic directly in the ConnectorCard component, since that is where the Formik form is defined, so having the form submission logic live close to the form itself might be nice. 

If this change is made, you may even be able to get rid of the useConnectorCardService hook, and just keep the useUIWidget hook, since that one still seems useful for encapsulating the uiWidget logic.

Definitely open to feedback here though if anyone disagrees - I'm still trying to learn what the best practices are around when and how exactly hooks should be used.